qcodes/tests/parameter/test_parameter_cache.py

```diff
@@ -330,3 +330,29 @@ def test_set_latest_works_for_plain_memory_parameter(p, value, raw_value):
     # Assert latest value and raw_value via private attributes for strictness
     assert p.cache._value == value
     assert p.cache._raw_value == raw_value
+
+
+def test_get_from_cache_marked_invalid():
+    param = BetterGettableParam(name="param", max_val_age=1)
+    param.get()
+    assert param._get_count == 1
+
+    param.cache.get(get_if_invalid=False)
+    assert param._get_count == 1
+
+    param.cache.invalidate()
+
+    param.cache.get(get_if_invalid=True)
+    assert param._get_count == 2
+
+    param.cache.invalidate()
+
+    param.cache.get(get_if_invalid=False)
+    assert param._get_count == 2
```

Comment:
Is this really the behaviour we want. Should this raise or warn?

Reply:
are you asking if we should differentiate between the explicit invalidation via `invalidate` call and other invalidations like the max_val_age? i'd say that for the clean-ness of the interface, if cache is invalid, it is simply invalid - hence i agree with this test.